src/env/environment.c

```diff
@@ -0,0 +1,101 @@
+#include <env/environment.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <libft.h>
+
+t_bool	export(t_env *env, const char *key_value_str)
+{
+	char	key_buffer[4096];
+	int		i;
+
+	if (!copy_key_to_buffer(key_value_str, key_buffer))
+		return (FALSE);
+	i = 0;
+	while (i < ENV_SIZE && env[i].key)
```

Comment:
An idea to do the loop for `i` or size into a function like `get_next_available_index()`?

Also `i` will always be smaller than `ENV_SIZE` as that's one of the conditions in the loop.

Reply:
Yeah, makes sense!
I'll leave this PR hanging until I'm back, then I can change it 😃

Reply:
Already added this change.

Reply:
> Very clear and readable! I added two comments one in enviroment.c and one in the utils.c.
> 
> Some small things i noticed:
> 
> * the "=" could be added to the e_special_chars.
> * `ft_strchr(key_value_str, '=')` gets used more than once, could be thrown into a helper function, `get_equal_sign()`.
> 
> Still not over how cool the " strchr(og_string) - &og_string[0] " trick is! Nois!

I made a define for `EQUAL_SIGN`, don't think it makes sense to add it to `special_chars` because `special_chars` will be used when parsing the input to find things like `| << >> `, things that split commands. Added a function `find_equal_sign_position()`

Reply:
Yeeep that makes a lot of sense, didn't think of the ENV_SIZE in that way!
Looks good Monique!